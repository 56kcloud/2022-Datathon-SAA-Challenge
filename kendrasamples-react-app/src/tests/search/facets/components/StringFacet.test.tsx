import React from "react";
import { shallow } from "enzyme";
import { StringFacet } from "../../../../search/facets/components/StringFacet";
import { mockDocumentAttributeValueCountPairs } from "../../../mock/mock";
import _ from "lodash";

const facetConfiguration = {
  facetsToShowWhenUncollapsed: 5,
  showCount: true,
};

describe("A <StringFacet />", () => {
  it("renders correctly", () => {
    const component = shallow(
      <StringFacet
        attributeName="mockAttributeName"
        valueCountPairs={mockDocumentAttributeValueCountPairs}
        disableAdd={false}
        facetConfiguration={facetConfiguration}
        onSelectionChange={_.noop}
        onClear={_.noop}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
