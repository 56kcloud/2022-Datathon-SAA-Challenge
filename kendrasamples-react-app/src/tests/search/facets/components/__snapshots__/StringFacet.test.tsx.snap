// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <StringFacet /> renders correctly 1`] = `
<Facet
  attributeName="mockAttributeName"
  disableAdd={false}
  facetConfiguration={
    Object {
      "facetsToShowWhenUncollapsed": 5,
      "showCount": true,
    }
  }
  onClear={[Function]}
  onSelectionChange={[Function]}
  values={
    Array [
      Object {
        "Count": 10,
        "DocumentAttributeValue": Object {
          "StringValue": "mockStringValue1",
        },
        "ValueLabel": "mockStringValue1",
      },
      Object {
        "Count": 20,
        "DocumentAttributeValue": Object {
          "StringValue": "mockStringValue2",
        },
        "ValueLabel": "mockStringValue2",
      },
    ]
  }
/>
`;
